pkg/resource/predicates.go

```diff
@@ -135,16 +137,13 @@ func IsPropagator() PredicateFn {
 		}
 
 		a := ao.GetAnnotations()
-		switch {
-		case a[AnnotationKeyPropagateToNamespace] == "":
-			return false
-		case a[AnnotationKeyPropagateToName] == "":
-			return false
-		case a[AnnotationKeyPropagateToUID] == "":
-			return false
-		default:
-			return true
+		for key := range a {
```

Comment:
Nit: You could probably iterate over `ao.GetAnnotations()` directly now that this function uses it only once.